### ISSUE TITLE
 Assistance with Saving Cropped Images and Classification with YOLOv8

### ISSUE THREAD

**Issue by cabral0413** [ultralytics/ultralytics#14844]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Dear YOLOv8 Team,

I am working on a snake classification project and using YOLOv8 for both object detection and classification tasks. I have encountered a couple of questions and would appreciate your guidance.

1. Cropping and Saving Images:

I am currently using YOLOv8 for object detection with the following code snippet to crop and save images:

```

from ultralytics import YOLO
import torch

model = YOLO('/content/drive/MyDrive/YolodefFinal/runs/detect/train3/weights/best.pt')

# Define path to video file
source = '/content/GV.jpeg'

# Run inference on the source
results = model.predict(source, show=True, save=True, save_crop=True)  # generator of Results objects
```
This code successfully crops images and saves them in the runs/detect folder. However, I would like to save these cropped images in a different folder within my project directory. How can I modify this process to specify a custom save directory for cropped images?

2. Classification Process:

In the classification phase, I use the following code to perform classification:

```

from flask import Flask, request
from ultralytics import YOLO
from PIL import Image

app = Flask(__name__)

@app.route('/classify', methods=['POST'])
def predict():
    file = request.files['image']
    img = Image.open(file.stream).convert("RGB")  # Ensure it's in RGB format
    
    model = YOLO('models/best.pt')  # Make sure the path is correct
    results = model(img)  # Directly pass PIL image

    # Check if any detections are made
    if len(results) == 0:
        return {'message': 'No objects detected in the image.'}
    
    # Assuming we're doing classification, let's find the top two classes if available
    if len(results) >= 2:
        top_classes = [(results[i].names[results[i].probs.top1], results[i].probs.top1conf.item()) for i in range(2)]
    else:
        top_classes = [(results[i].names[results[i].probs.top1], results[i].probs.top1conf.item()) for i in range(len(results))]
    
    # Determine venom status for each top class
    top_venom_status = [get_venom_status(class_name) for class_name, _ in top_classes]
    
    # Format the probabilities as percentages with two decimal points
    formatted_probs = ["{:.2%}".format(confidence) for _, confidence in top_classes]

    # Check if confidence is below 70% for any top class
    if any(confidence < 0.7 for _, confidence in top_classes):
        return {'message': 'No snakes detected with sufficient confidence. Please upload a clearer photo.'}

    return {'top_predictions': [{'class': class_name, 'probability': prob, 'venom_status': venom_status}
                                for (class_name, _), prob, venom_status in zip(top_classes, formatted_probs, top_venom_status)]}

def get_venom_status(class_name):
    # Implement your logic to determine venom status based on the class name
    if class_name == 'Common Indian Krait':
        return 'Venomous'
    elif class_name == 'Python':
        return 'Venomous'
    elif class_name == 'Hump Nosed Viper':
        return 'Venomous'
    elif class_name == 'Green Vine Snake':
        return 'Non-venomous'
    elif class_name == 'Russells Viper':
        return 'Venomous'
    elif class_name == 'Indian Cobra':
        return 'Venomous'
    else:
        return 'Unknown'

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5000, debug=True)
```
I would like to understand how to first save the cropped images from YOLOv8 into a specific directory within my project, and then how to load these images for classification using a separate YOLOv8 model.

Thank you for your assistance.

Best regards,
Nipuni Cabral

**Comment by Y-T-G** [comment]
You can crop it using the bbox coordinates:
https://docs.ultralytics.com/modes/predict/#working-with-results

Also you shouldn't be loading the model everytime you want to predict. It should be loaded once in the beginning of the script. And I don't understand what is the need for saving it to the disk and loading it again, and the need for two different scripts. That's unnecessary and slow I/O. You can just pass the cropped image array to the classification model within the same script. There's no need to save them in between.

**Comment by cabral0413** [comment]
I appreciate the information on using bounding box coordinates for cropping. I initially saved the cropped images to verify their correctness. Once I confirm that the cropping is working as expected, I plan to remove the saving step and directly pass the cropped image arrays to the classification model.

**Comment by ajay2177** [comment]
Hi @cabral0413, as @Y-T-G mentioned, it is better to combine both the scripts into one. If you still want to save the images to a directory of your own, you need to make changes to several functions in the Ultralytics Project. But an easy workaround is to use shutil module to move the images to your required directory after they are saved in the default directory. Also from your Classification Process code, I can see you are iterating over the results which would contain the results of a batch of images provided to the model as input. If you are passing only one image, then you need to index the first and only element in the results variable like this 'result = results[0]'. Then you can access its attributes.

**Comment by cabral0413** [comment]
hi @ajay2177 ,

Thank you for the suggestions and insights.

I am developing a mobile app to detect and classify snakes. When a user uploads an image to the app, I need to classify the image and display the results.

I agree that combining the object detection and classification scripts would be more efficient. However, when I attempted to merge them, I encountered errors. This is why I asked for help.

For now, I will implement the suggested workaround using the shutil module to move images to the desired directory after they are saved in the default directory. I will also adjust the indexing in the predict function to handle a single image correctly.

Thank you for your guidance.
